packages/docusaurus-theme-common/src/hooks/useShallowMemoObject.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useMemo} from 'react';
+
+export default function useShallowMemoObject<O>(obj: O): O {
```

Comment:
Please put this in `reactUtils`. The `hooks` folder is for untestable hooks that interact with DOM or storage